### PR TITLE
fix: CVR GND CTRL

### PR DESCRIPTION
**Pull request by tracernz**
## Summary of Changes

- CVR ground control automatically resets upon first engine start, as requested by IRL A320 pilot FlyingCarpet7, and is also now a momentary push to toggle.

## References

[YouTube for making it momentary](https://www.youtube.com/watch?v=ceDKP1qnWwg).
FCOM and other docs for logic.

## Additional context

Was previously part of #4997 

This does not implement any CVR logic except the manual recording mode. Implementing any other part would not make sense as the result would not be visible to the pilot in any way.

Discord username (if different from GitHub):

## Testing instructions

With no engines running:
- Pressing and holding GND CTL button turns GND CTL on with initial press, releasing has no further effect
- Pressing and holding GND CTL button has no effect... releasing turns GND CTL off
- Press and release GND CTL to turn it on
- Press CVR TEST button and make sure you hear a beep
- Start engine one and ensure GND CTL switches off when the oil pressure rises
- Press GND CTL and ensure it cannot be enabled while the engine is running
- Shut the engine down and repeat for engine two

General check of cockpit buttons for any regression.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js [hunk 1]
```diff
@@ -272,5 +278,14 @@ class A32NX_LocalVarUpdater {
         return this.opacityMcdu;
     }
 
+    cvrGroundControl(deltaTime, identifier, currentValue) {
+        if (currentValue) {
+            const isEngineOneRunning = SimVar.GetSimVarValue("ENG COMBUSTION:1", "bool");
+            const isEngineTwoRunning = SimVar.GetSimVarValue("ENG COMBUSTION:2", "bool");
+            return !isEngineOneRunning && !isEngineTwoRunning;
+        }
+        return currentValue;
```
Comment (bouveng) [inline flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js:287]: When will `currentValue` be returned?

**Review by bouveng — DISMISSED** [review]

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js (hunk 1 above)
Comment (tracernz) [reply flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js:287]: Standby, I am re-implementing this with a better reference. ;)

**Review by bouveng — APPROVED** [review]

**Comment by John-UK** [comment]
Quality Assurance Tester

Discord: john P Maguire#9027
Object of testing: #5683 
Date: 19/08/2021

Build successful (15 mins)
Testing Process:
Started C&D at a stand, completed all the flows including selecting GND CTR to on and testing CVR tone.
Started engine 2 at approximately 35 PSI the switch light extinguished. Passed
Repeated the process for starting engine 1 first. Passed.
Shut down each engine in turn. The GND CTR cannot be switched on when one engine running. Passed
Shut down remaining engine. The GND CTR cannot be switched on when engine shutting down.
The GND CTL can be switched on immediately after the last engine running is shut down. There is a time delay of 5 mins after the engine shut down where the ground recorder is running.  To be implemented? 
Switched engines off for more than five mins (on APU) the light did not revert to on. Passed.


Negatives: Time delay on selecting to on after final engine is shut down
Testing Results: Passed
Conclusions: This used to annoy me every time I looked up after starting so nice to see the light off.

**Comment by Highflyer2010** [comment]
Quality Assurance: Trainee Report

Discord:Highflyer2010#8932
Object of testing: #5683
Tier of Testing: 1
Date: (19/08/2021)

Testing Process:
   [Engine 1]
    (With no engines running, Cold and Dark at EGNT Gate 10)
    *Pressing and holding GND CTL button turns GND CTL on with initial press, releasing has no further effect / PASSED
    *Pressing and holding GND CTL button has no effect... releasing turns GND CTL off / PASSED
    *Press and release GND CTL to turn it on / FAILED - Quick press
    *Press CVR TEST button and make sure you hear a beep / PASSED
    *Start engine one and ensure GND CTL switches off when the oil pressure rises / PASSED
    *Press GND CTL and ensure it cannot be enabled while the engine is running / PASSED
    *Shut the engine down.

  [Engine 2]
    (With no engines running, Cold and Dark at EGNT Gate 10)
    *Pressing and holding GND CTL button turns GND CTL on with initial press, releasing has no further effect / PASSED
    *Pressing and holding GND CTL button has no effect... releasing turns GND CTL off / PASSED
    *Press and release GND CTL to turn it on / FAILED - Quick pressing the GND CTL does not switch it on
    *Press CVR TEST button and make sure you hear a beep / PASSED
    *Start engine one and ensure GND CTL switches off when the oil pressure rises / PASSED
    *Press GND CTL and ensure it cannot be enabled while the engine is running / PASSED
    *Shut the engine down 

General check of cockpit buttons for any regression.

Negatives: 
I've noticed press Quick double clicking on the GRD CTL button would not turn on. But this is happening with all buttons within the cockpit so maybe this could be a setting within MSFS or it's ment to be like that I've giving this PR a Pass for that reason as 
if it was only the one button then i would of failed it. I've Attatched a video for you to see. When the Hand Cursor Dissappears in the video is when it's happening.

Testing Results: Passed

Conclusions: (Notes made, the positives in the PR, and anything extra you would like to mention - Delete Line if Nothing Extra is to be mentioned)
Media: [https://files.fm/f/5b569peub](url)

File: src/behavior/src/A32NX_Interior_Generics.xml [hunk 2]
```diff
@@ -417,12 +418,28 @@
...
-                        <LEFT_SINGLE_CODE>#LEFT_SINGLE_CODE#</LEFT_SINGLE_CODE>
-                        <LEFT_LEAVE_CODE>#LEFT_SINGLE_CODE#</LEFT_LEAVE_CODE>
+                        <Condition Valid="LEFT_LEAVE_CODE">
+                            <True>
+                                <LEFT_SINGLE_CODE>#LEFT_SINGLE_CODE#</LEFT_SINGLE_CODE>
+                                <LEFT_LEAVE_CODE>#LEFT_LEAVE_CODE#</LEFT_LEAVE_CODE>
+                            </True>
+                            <False>
+                                <LEFT_SINGLE_CODE>#LEFT_SINGLE_CODE#</LEFT_SINGLE_CODE>
+                                <LEFT_LEAVE_CODE>#LEFT_SINGLE_CODE#</LEFT_LEAVE_CODE>
```
Comment (Benjozork) [inline src/behavior/src/A32NX_Interior_Generics.xml:428]: How does this not break toggling simvars ? Wouldn't `LEFT_SINGLE_CODE` run every time the button is released ?

File: src/behavior/src/A32NX_Interior_Misc.xml [hunk 3]
```diff
@@ -647,4 +647,52 @@
...
+
+// emulate 12KS1 and 12KS2
+alias eng1OilPressure = (A:GENERAL ENG OIL PRESSURE:1, psi);
+alias eng2OilPressure = (A:GENERAL ENG OIL PRESSURE:2, psi);
+let lowOilPressureThreshold = 18;
+let eng1Off = eng1OilPressure &lt; lowOilPressureThreshold;
+let eng2Off = eng2OilPressure &lt; lowOilPressureThreshold;
+
+// permissives for relay 12TU
+ctlPower and eng1Off and eng2Off
```
Comment (Benjozork) [inline src/behavior/src/A32NX_Interior_Misc.xml:665]: Merged with broken formatting ...

File: src/behavior/src/A32NX_Interior_Generics.xml (hunk 2 above)
Comment (tracernz) [reply src/behavior/src/A32NX_Interior_Generics.xml:428]: Check the existing code, exactly the same..

**Pull request merged**